src/snakemake/common/__init__.py

```
@@ -293,7 +293,8 @@ def get_function_params(func: Callable):
     if hasattr(func, FUNC_OVERWRITE_PARAMS_ATTR):
         return getattr(func, FUNC_OVERWRITE_PARAMS_ATTR)
     else:
-        return inspect.signature(func).parameters
+        callable_func = func.callable if is_flagged(func, "unpack") else func
```

Ruff (0.14.0)

Undefined name `is_flagged`

(F821)